### PR TITLE
src: Add completion for the Kakoune code editor

### PR DESCRIPTION
**Pull request by lenormf**
Hi,

This is a completion file for [Kakoune](https://github.com/mawww/kakoune), all the options available in the mainstream `kak` binary are implemented here.

Let me know if changes are needed.

Thanks.

### CODE REVIEW

File: src/_kak [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# vim: ft=zsh sw=2 ts=2 et
+# ------------------------------------------------------------------------------
+
+_kak_sessions() {
+  session_ids=""
+  for session_id in $(kak -l); do
+    session_ids="${session_ids} '${session_id}'"
+  done
+
+  eval "_values ${session_ids}"
```
Comment (nicoulaj) [inline src/_kak]: I don't see why you would use eval here, can you submit another PR to remove it ?

File: src/_kak [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+#
+#  * Frank LENORMAND <https://github.com/lenormf>
+#
+# ------------------------------------------------------------------------------
+# -*- mode: zsh; sh-indentation: 2; indent-tabs-mode: nil; sh-basic-offset: 2; -*-
+# vim: ft=zsh sw=2 ts=2 et
+# ------------------------------------------------------------------------------
+
+_kak_sessions() {
+  session_ids=""
```
Comment (nicoulaj) [inline src/_kak]: The proper way to do this is to use `_call_program`, there are [plenty of examples in the project](https://github.com/zsh-users/zsh-completions/search?q=_call_program).

File: src/_kak [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  eval "_values ${session_ids}"
+}
+
+_kak() {
+  _arguments \
+        '-n[do not source rc files on startup]' \
+        '-l[list existing sessions]:session_id:_kak_sessions' \
+        '-clear[clear dead sessions]' \
+        '-ro[readonly mode]' \
+        '-help[display a help message and quit]' \
```
Comment (nicoulaj) [inline src/_kak]: You can do it in one line using expansion: `{-help,--help}`, see other compdefs.

Also, I believe you can tell `_arguments` that long option with a single hyphen are supported, but I don't remember how.

**Comment by nicoulaj** [comment]
I merged despite the comments, feel free to send another PR if you would like to fix it.

**Comment by lenormf** [comment]
Thanks for taking the time to comment, I will make another PR if I ever get around to fixing this file.

File: src/_kak (hunk 1 above)
Comment (lenormf) [inline src/_kak]: I automatically use `eval` whenever I generate a command dynamically, I guess it can go.
